### ISSUE TITLE
Abort at Outro (the "The End" screen before the credits)

### ISSUE THREAD

**Issue by doofus-01** [wesnoth/wesnoth#4960]
### Game and System Information
The Battle for Wesnoth version 1.15.3+dev (1ff6b7bbc4a-Modified)
Running on openSUSE Leap 15.1
Distribution channel: Default
#### Libraries
Boost:             1.66
OpenSSL/libcrypto: 1.1.0i (runtime 1.1.0i)
Cairo:             1.15.10 (runtime 1.15.10)
Pango:             1.40.13 (runtime 1.40.13)
SDL:               2.0.8 (runtime 2.0.8)
SDL_image:         2.0.5 (runtime 2.0.5)
SDL_mixer:         2.0.2 (runtime 2.0.2)
SDL_ttf:           2.0.14 (runtime 2.0.14)
#### Current video settings
SDL video drivers:       [x11] wayland KMSDRM dummy
Window size:             1920x1015
Screen refresh rate:     60
Screen dots per inch:    128.000000x128.186920
Screen dpi scale factor: 1.333333x1.335280

### Describe the bug
Game aborts at the end of campaign (UtBS), I think it's related to the credits screen.  The window size or full-screen do not seem to matter.
```
wesnoth: src/gui/widgets/window.cpp:1005: void gui2::window::layout(): Assertion `size.x > 0 && static_cast<unsigned>(size.x) <= maximum_width && size.y > 0 && static_cast<unsigned>(size.y) <= maximum_height' failed.
Aborted (core dumped)
```

### To Reproduce

Use the attached save-file, click through to the end.
[UtBS-Epilogue_20200621-103309.gz](https://github.com/wesnoth/wesnoth/files/4810117/UtBS-Epilogue_20200621-103309.gz)

### Expected behavior

Go to credits.

**Comment by Konrad22** [comment]
Likely related or even a duplicate of #3265

**Comment by doofus-01** [comment]
That one was closed.

**Comment by Konrad22** [comment]
I can reproduce it on 1.15.3, so I think it should be reopened.

**Comment by Pentarctagon** [comment]
Given I've re-opened the other one, should this be closed then?

**Comment by doofus-01** [comment]
If you're sure they are the same bug, but aside from the fact they both seem to deal with the credits, I wouldn't have guessed they were the same.  The errors are different, this one doesn't need any add-ons installed, and clicking on the credits button doesn't cause any problem here.

**Comment by Wedge009** [comment]
Confirmed this on new master build today (1.15.3+dev at acd08fd0c51) - I just started a new UtBS campaign, debug-choose_level to Epilogue, go through the story scene and where the credits should appear, Wesnoth crashes and reports the same error doofus did.

Doing the same on 1.14.13 release does not have this issue.

**Comment by stevecotton** [comment]
Before the credits are shown, there's a screen saying "The End". It seems this crash is in that screen, and unrelated to the credits.

That screen is outro.cpp, outro.hpp and outro.cfg, however none of these files have an obvious recent change to retest. I'm running a bisection to find out when it broke.

**Comment by stevecotton** [comment]
@gwhitney , this seems to be a bug introduced in 49d8779ff1aadd753762bff57a57b58476989470. One of the asserts changed in that commit now triggers when get_best_size() returns [0,0], however the outro screen's implementation is unchanged and triggers the new assertion.

Wedge009's https://github.com/wesnoth/wesnoth/issues/4960#issuecomment-652779425 gives the easiest way to reproduce it.

**Comment by gwhitney** [comment]
Apologies, will look & fix this weekend. Thanks for the alert!

**Comment by Wedge009** [comment]
Thanks for finding and attending to this!

**Comment by Wedge009** [comment]
I just did a quick test of a few other campaigns and it doesn't appear to be specific to UtBS, looks like it's the credits in general - might be obvious but I'll just throw that in there just in case. (If I recall correctly there may be campaign-specific credits as well as the general credits at the end.)

**Comment by gwhitney** [comment]
Sorry about the delay! I was just waiting so I could snag a nice round pull request number ;-) No, seriously, I had a very pressing project come down that Friday which pretty much consumed all my time, so I couldn't take a look until now. And then I felt sheepish, because I was just being overly aggressive when I fixed the type mismatch in that assert; I added in the non-zero checks, assuming that no window could actually want a zero size, but I was just blindly assuming and turns out I was wrong. My apologies.
